tests/test_transforms/test_math/test_sklearn_transform_interface.py

```diff
@@ -278,3 +278,319 @@ def test_ordering(transform_constructor, in_column, mode, multicolumn_ts):
+@pytest.mark.parametrize(
+    "in_column",
+    [
+        "exog_1",
+        ["exog_1", "exog_2"],
+    ],
+)
+@pytest.mark.parametrize(
+    "transform_constructor",
+    [
+        BoxCoxTransform,
+        YeoJohnsonTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+        MaxAbsScalerTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+    ],
+)
+def test_transform_new_segments_per_segment_fail(transform_constructor, in_column, inplace, multicolumn_ts):
+    df = multicolumn_ts.to_pandas()
+    train_df = df.loc[:, pd.IndexSlice[["segment_0", "segment_1"], :]]
+    test_df = df.loc[:, pd.IndexSlice["segment_2", :]]
+    transform = transform_constructor(mode="per-segment", in_column=in_column, inplace=inplace)
+
+    transform.fit(train_df)
+    with pytest.raises(
+        NotImplementedError, match="This transform can't process segments that weren't present on train data"
+    ):
+        _ = transform.transform(test_df)
+
+
+@pytest.mark.parametrize("inplace", [False, True])
+@pytest.mark.parametrize(
+    "in_column",
+    [
+        "exog_1",
+        ["exog_1", "exog_2"],
+    ],
+)
+@pytest.mark.parametrize(
+    "transform_constructor",
+    [
+        BoxCoxTransform,
+        YeoJohnsonTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+        MaxAbsScalerTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+    ],
+)
+def test_inverse_transform_new_segments_macro(transform_constructor, in_column, inplace, multicolumn_ts):
+    df = multicolumn_ts.to_pandas()
+    train_df = df.loc[:, pd.IndexSlice[["segment_0", "segment_1"], :]]
+    test_df = df.loc[:, pd.IndexSlice["segment_2", :]]
+    transform = transform_constructor(mode="macro", in_column=in_column, inplace=inplace)
+
+    transform.fit(train_df)
+    _ = transform.inverse_transform(test_df)
+
+
+@pytest.mark.parametrize(
+    "in_column",
+    [
+        "exog_1",
+        ["exog_1", "exog_2"],
+    ],
+)
+@pytest.mark.parametrize(
+    "transform_constructor",
+    [
+        BoxCoxTransform,
+        YeoJohnsonTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+        MaxAbsScalerTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+    ],
+)
+def test_inverse_transform_new_segments_per_segment_non_inplace(transform_constructor, in_column, multicolumn_ts):
+    df = multicolumn_ts.to_pandas()
+    train_df = df.loc[:, pd.IndexSlice[["segment_0", "segment_1"], :]]
+    test_df = df.loc[:, pd.IndexSlice["segment_2", :]]
+    transform = transform_constructor(mode="per-segment", in_column=in_column, inplace=False)
+
+    transform.fit(train_df)
+    _ = transform.inverse_transform(test_df)
```

Comment:
So, here, `segment_2` should be left untouched after applying `inverse_transform`, isn't it?
If so, can we test this behaviour explicitly by comparing data frames?

Reply:
Fixed